### PR TITLE
Feat: update v2.1.0

### LINKED ISSUE #63: feat: update listings to v2.1.0
Update all listings to [v2.1.0](https://community.starknet.io/t/cairo-v2-1-0-is-out/98136) by changing the starknet dependencies in the `Scarb.toml` and make the relevant changes if needed.

### PR DESCRIPTION
**Pull request by julio4**
Close #63 #64

### CODE REVIEW

File: listings/ch01-advanced-concepts/storing_arrays/src/storing_arrays.cairo [hunk 1]
```diff
@@ -73,26 +69,22 @@ impl StorageAccessFelt252Array of StorageAccess<Array<felt252>> {
         }
     }
 
-    fn size_internal(value: Array<felt252>) -> u8 {
-        if value.len() == 0 {
-            return 1;
-        }
-        1_u8 + StorageAccess::<felt252>::size_internal(*value[0]) * value.len().try_into().unwrap()
+    fn size() -> u8 {
```
Comment (julio4) [inline listings/ch01-advanced-concepts/storing_arrays/src/storing_arrays.cairo:72]: @enitrat Do you know how to upgrade this part to 2.1.0?
The `size` function don't take a `value` arg anymore and the size is fixed by type.

Comment (enitrat) [reply listings/ch01-advanced-concepts/storing_arrays/src/storing_arrays.cairo:72]: yeah this is a bit troublesome, I can't think of any workaround for this. Perhaps `size` could return the maximum value, so `255`?

Comment (julio4) [reply listings/ch01-advanced-concepts/storing_arrays/src/storing_arrays.cairo:72]: I agree that returning the upper bound is a solution. This is however a bit confusing with an `array` type.

File: scripts/cairo_programs_format.sh
[the remark this answers is not in the document]
Comment (julio4) [reply scripts/cairo_programs_format.sh:1]: Yes, I added this as an helper to automatically run `scarb fmt` in all directory. And `scarb fmt -c` is run in CI.

File: listings/ch01-advanced-concepts/storing_arrays/src/storing_arrays.cairo (hunk 1 above)
Comment (enitrat) [reply listings/ch01-advanced-concepts/storing_arrays/src/storing_arrays.cairo:72]: this question has been asked quite a lot and I don't think we received an answer from the compiler team

File: scripts/cairo_programs_format.sh
[the remark this answers is not in the document]
Comment (enitrat) [reply scripts/cairo_programs_format.sh:1]: gotcha, nice

**Review by enitrat — APPROVED** [review]

**Pull request merged**